sas9.4-viya/scripts/ansible_setup.sh

```diff
@@ -0,0 +1,264 @@
+#!/bin/bash
+set -x
+echo "*** Phase 1 - Ansible Startup Script Started at `date +'%Y-%m-%d_%H-%M-%S'` ***"
+
+##Error handling functions
+fail_if_error() {
+  [ $1 != 0 ] && {
+    echo $2
+    exit 10
+  }
+}
+
+fail_if_error_phrase() {
+  [ $1 != 0 ] && {
+    unset PASSPHRASE
+    exit 10
+  }
+}
+
+## Facter Installation For Dynamic Values
+yum install ruby wget -y
+wget http://download-ib01.fedoraproject.org/pub/epel/7/x86_64/Packages/f/facter-2.4.1-1.el7.x86_64.rpm
+rpm  --reinstall -vh facter-2.*.rpm
+fail_if_error $? "Facter Installation failed"
+
+#Facter Variables and Declaration
+environmentLocation="/etc/facter/facts.d/variables.txt"
+
+if [ -f $environmentLocation ]; then
+	rm -f $environmentLocation
+fi
+
+cat << EOF > $environmentLocation
+azure_storage_account=$1
+azure_storage_files_share=$2
+viyarepo_folder=$3
+app_name=$4
+domain_name=$5
+ansible_vmname=$6
+microservices_vmname=$7
+cascontroller_vmname=$8
+spre_vmname=${9}
+casworker_vmname=${10}
+saspwd=${11}
+caspwd=${12}
+kv_vault_name=${13}
+secret_pvt_keyname=${14}
+secret_pub_keyname=${15}
+cas_nodes=${16}
+mid_name=${17}
+artifact_loc=${18}
+compute_name=${19}
+meta_name=${20}
+stgacc_secr_name=${21}
+EOF
+
+#Defining variables with facter values
+azure_storage_account=`facter azure_storage_account`
+cifs_server_fqdn="${azure_storage_account}.file.core.windows.net"
```

Comment:
remove hard-coded endpoint (pass in the FQDN)

Reply:
since we are taking the storage account name as a parameter from the end-user. I could find a way to send FQDN without concating the endpoint. We are using this storage account file share to mount on the VM's.

Reply:
see the environment() function, e.g. 

concat(storageAccountName, '.file.', environment().suffixes.storage)